aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -28,7 +28,8 @@ static inline std::tuple<Tensor, Tensor> _lu_det_P_diag_U(const Tensor& self) {
   std::tie(lu, pivs, infos) = at::_lu_with_info(self, /*pivot=*/true, /*check_errors=*/false);
   TORCH_CHECK(infos.ge(0).all().item<uint8_t>(), "Invalid argument passed to lu");
   auto n = self.size(-1);
-  auto num_exchanges = (at::arange(1, n + 1, pivs.options()) != pivs).sum(-1, /*keepdim=*/false, /*dtype=*/self.scalar_type()).fmod_(2);
+  auto num_exchanges = (at::arange(1, n + 1, pivs.options().dtype(at::kLong)) != pivs.to(at::kLong))
+    .sum(-1, /*keepdim=*/false, /*dtype=*/at::kLong).fmod_(2);
   // NB: the `.contiguous()` call is added due to the bug in `.prod()` as reported in
   // issue #https://github.com/pytorch/pytorch/issues/34061
```

Comment:
#34061 looks to be fixed, should we remove this workaround?

Reply:
to @ngimel 's point, let's remove this workaround since #34061 is fixed